packages/shared/src/components/cards/entity/SourceEntityCard.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useMemo } from 'react';
+import { useRouter } from 'next/router';
+import EntityCard from './EntityCard';
+import {
+  Typography,
+  TypographyColor,
+  TypographyType,
+} from '../../typography/Typography';
+import type { Source, SourceTooltip } from '../../../graphql/sources';
+import { LogEvent } from '../../../lib/log';
+import { ContentPreferenceType } from '../../../graphql/contentPreference';
+import { largeNumberFormat, ReferralCampaignKey } from '../../../lib';
+import CustomFeedOptionsMenu from '../../CustomFeedOptionsMenu';
+import { useContentPreference } from '../../../hooks/contentPreference/useContentPreference';
+import SourceActionsFollow from '../../sources/SourceActions/SourceActionsFollow';
+import { ButtonVariant } from '../../buttons/Button';
+import { useSourceActions } from '../../../hooks';
+import { Separator } from '../common/common';
+import EntityDescription from './EntityDescription';
+
+const SourceEntityCard = ({ source }: { source: SourceTooltip }) => {
+  const { isFollowing, toggleFollow } = useSourceActions({
+    source: source as Source,
+  });
+  const router = useRouter();
+  const { follow, unfollow } = useContentPreference();
+
+  const actionButtons = useMemo(() => {
+    return (
+      <>
+        <CustomFeedOptionsMenu
+          className={{
+            button: 'bg-background-popover',
+            menu: 'z-[9999]',
+          }}
+          onCreateNewFeed={() =>
+            router.push(
+              `/feeds/new?entityId=${source.id}&entityType=${ContentPreferenceType.Source}`,
```

Comment:
Also doesn't this need pre-added webappUrl, pretty sure you could see these in extension right?